airbyte-integrations/connectors/source-google-ads/unit_tests/test_incremental_events_streams.py

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from unittest.mock import Mock
+
+import pytest
+from airbyte_cdk.models import SyncMode
+from source_google_ads.google_ads import GoogleAds
+from source_google_ads.streams import CampaignCriterion, ChangeStatus
+
+from .common import MockGoogleAdsClient as MockGoogleAdsClient
+
+
+@pytest.fixture
+def mock_ads_client(mocker, config):
+    """Mock google ads library method, so it returns mocked Client"""
+    mocker.patch("source_google_ads.google_ads.GoogleAdsClient.load_from_dict",
+                 return_value=MockGoogleAdsClient(config))
+
+
+def mock_response_parent():
+    yield [
+        {"change_status.last_change_date_time": "2023-06-13 12:36:01.772447",
+         "change_status.resource_type": "CAMPAIGN_CRITERION", "change_status.resource_status": "ADDED",
+         "change_status.campaign_criterion": "1"},
+        {"change_status.last_change_date_time": "2023-06-13 12:36:02.772447",
+         "change_status.resource_type": "CAMPAIGN_CRITERION", "change_status.resource_status": "ADDED",
+         "change_status.campaign_criterion": "2"},
+        {"change_status.last_change_date_time": "2023-06-13 12:36:03.772447",
+         "change_status.resource_type": "CAMPAIGN_CRITERION", "change_status.resource_status": "REMOVED",
+         "change_status.campaign_criterion": "3"},
+        {"change_status.last_change_date_time": "2023-06-13 12:36:04.772447",
+         "change_status.resource_type": "CAMPAIGN_CRITERION", "change_status.resource_status": "REMOVED",
+         "change_status.campaign_criterion": "4"},
+    ]
+
+
+def mock_response_child():
+    yield [
+        {"customer.id": 123, "campaign.id": 1, "campaign_criterion.resource_name": "1"},
+        {"customer.id": 123, "campaign.id": 1, "campaign_criterion.resource_name": "2"},
+    ]
+
+
+class MockGoogleAds(GoogleAds):
+    def parse_single_result(self, schema, result):
+        return result
+
+    def send_request(self, query: str, customer_id: str):
+        if query == "query_parent":
+            return mock_response_parent()
+        else:
+            return mock_response_child()
+
+
+def test_change_status_stream(mock_ads_client, config, customers):
+    """
+    """
+    customer_id = next(iter(customers)).id
+    stream_slice = {"customer_id": customer_id}
+
+    google_api = MockGoogleAds(credentials=config["credentials"])
+
+    stream = ChangeStatus(api=google_api, customers=customers)
+
+    stream.get_query = Mock()
+    stream.get_query.return_value = "query_parent"
+
+    result = list(
+        stream.read_records(sync_mode=SyncMode.incremental, cursor_field=["change_status.last_change_date_time"],
+                            stream_slice=stream_slice))
+    assert len(result) == 4
```

Comment:
can you assert on the content of the records too?

Reply:
Fixed